### PR TITLE
fall back to k8s exit code if there is no termination log

### PR DESCRIPTION
**Pull request by danielnorberg**

### CODE REVIEW

File: styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesPodEventTranslator.java [hunk 1]
```diff
@@ -64,26 +67,46 @@ public TerminationLogMessage(
...
-        return Optional.of(message.exitCode);
-      } catch (IOException e) {
-        LOG.warn("Unexpected termination log message for container {}", status.getContainerID(), e);
+
+      // If there's no termination log exit code, fall back to k8s exit code if it is not zero.
+      // Rationale: It is important for users to be able to get the exit code of the container to be
+      // able to debug failures, but at the same time we must be careful about using the use the k8s
+      // exit code when checking whether the execution was successful as dockerd some times returns
+      // incorrect exit codes.
+      // TODO: consider separating execution status and debugging info in the "terminate" event.
```
Comment (danielnorberg) [inline styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesPodEventTranslator.java]: @rouzwawi wdyt about this?

File: styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesPodEventTranslator.java [hunk 2]
```diff
@@ -64,26 +69,46 @@ public TerminationLogMessage(
...
 
+    // Check termination log exit code, if available
     if ("true".equals(pod.getMetadata().getAnnotations().get(DOCKER_TERMINATION_LOGGING_ANNOTATION))) {
       if (terminated.getMessage() == null) {
         LOG.warn("Missing termination log message for container {}", status.getContainerID());
-        return Optional.empty();
+        stats.terminationLogMissing();
+      } else {
+        try {
+
```
Comment (honnix) [inline styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesPodEventTranslator.java]: empty line

File: styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesPodEventTranslator.java [hunk 3]
```diff
@@ -64,26 +69,46 @@ public TerminationLogMessage(
...
-      } catch (IOException e) {
-        LOG.warn("Unexpected termination log message for container {}", status.getContainerID(), e);
+
+      // If there's no termination log exit code, fall back to k8s exit code if it is not zero.
+      // Rationale: It is important for users to be able to get the exit code of the container to be
+      // able to debug failures, but at the same time we must be careful about using the use the k8s
+      // exit code when checking whether the execution was successful as dockerd some times returns
+      // incorrect exit codes.
+      // TODO: consider separating execution status and debugging info in the "terminate" event.
+      if (terminated.getExitCode() != null && terminated.getExitCode() != 0) {
```
Comment (honnix) [inline styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesPodEventTranslator.java]: `Integer.valueOf(0).equals(terminated.getExitCode())` ?

Comment (danielnorberg) [inline styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesPodEventTranslator.java]: `Objects.equals(0, terminated.getExitCode())`

**Pull request merged**